specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,428 @@
+
+* ApplicationData and ApplicationDataManager functionality
+* *Path properties
+* Reliable access
+* Per-Machine data store
+* Unpackaged app data stores
+
+## 3.1. ApplicationData and ApplicationDataManager functionality
+
+This API provides an equivalent superset of
+[Windows.Storage.ApplicationData](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata),
+with the following notable differences:
+
+* **No roaming functionality** -- We don't provide equivalents to Windows.Storage.ApplicationData's
+      [RoamingSettings](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingsettings),
+      [RoamingStorageQuota](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingstoragequota),
+      [Version](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.version),
+      [SetVersionAsync()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.setversionasync),
+      [SignalDataChanged()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.signaldatachanged)
+      and
+      [DataChanged](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.datachanged). See [5.1. Platform Support](#51-platform-support) for more details.
+* **GetDefault() works for all packaged apps** -- Equivalent to .Current but works for all packaged
+  apps (not just callers running in an AppContainer). See
+  [3.3. Reliable access](#33-reliable-access).
+* **`GetForPackageFamily()`** -- Equivalent to Windows.Storage.ApplicationDataManager.CreateForPackageFamily().
+* **`GetForUser()`** -- Synchronous equivalent to GetForUserAsync().
+* **`GetForUnpackaged()`** -- Equivalent functionality for unpackaged apps. See
+  [3.5. Unpackaged app data stores](#35-unpackaged-app-data-stores).
+
+## 3.2. *Path properties
+
+New *Path equivalents to Windows.Storage.ApplicationData's *Folder but returning a `String` rather than `StorageFolder`.
+
+## 3.3. Reliable access
+
+`Windows.Storage.ApplicationData.Current` only works for packaged apps **running in an AppContainer**. Processes not running in an AppContainer (e.g. MediumIL) must use the equivalent `Windows.Management.Core.ApplicationDataManager.CreateForPackageFamily(Windows.ApplicationModel.Package.Current.Id.FamilyName)`.
+
+Microsoft.Windows.Storage.ApplicationData provides consistent behavior as developers expect e.g. `ApplicationData.Current` works regardless if the caller's running in an AppContainer or not.
+
+## 3.4. Per-Machine data store
+
+Package families can optionally have a per-machine data store.
+
+### 3.4.1. Machine Path/Folder
+
+The path for a package family’s "Machine Folder" is located at `%ProgramData%\Microsoft\Windows\AppRepository\ApplicationData\...packagefamilyname...\Machine`.
+
+This directory is ACL’d similarly to a package’s System Metadata directory (`%ProgramData%\Microsoft\Windows\AppRepository\Packages\...pkgfullname...`) e.g.
+
+```
+BUILTIN\Users:(OI)(CI)(Rc,S,RD,REA,X,RA)
```

Comment:
FrancisH has come up with an interesting conditional ACE in an SDDL that grants access only when the caller is a certain package identity and running as a user. Handy way to cloak a directory from casual probing by other apps.

Reply:
Interesting. Color me curious; I'll contact him to learn more. Thanks for the tidbit.

Reply:
CONSIDER: Feature enhancement based on FrancisH example